cmd/main.go

```diff
@@ -1,53 +1,60 @@
 package main
 
 import (
-	"TicTacGo/internal/board"
-	"fmt"
+	"fmt" // first stdlib imports
+
+	// after empty line 3rd-party imports
+
+	// always use full import paths
+	// imternal package is needed only for packages that is explicitly forbidden to use from outside
+	"github.com/Kwbmm/TicTacGo/board" // after empty line local module imports
 )
 
 func main() {
 	fmt.Println("Welcome to TicTacGo")
 
 	currentPlayer := 'O'
 	nextPlayer := 'X'
-	board := board.Board{BoardState: [9]board.Cell{
-		board.Cell{Value: '1'},
-		board.Cell{Value: '2'},
-		board.Cell{Value: '3'},
-		board.Cell{Value: '4'},
-		board.Cell{Value: '5'},
-		board.Cell{Value: '6'},
-		board.Cell{Value: '7'},
-		board.Cell{Value: '8'},
-		board.Cell{Value: '9'},
-	}}
+
+	// it's better not to use variable name the same as package name
+	// too complex struct for simple task
+	var b board.Board
 
 	for {
-		board.PrintBoard()
+		b.Print()
+
 		fmt.Printf("Player '%c' turn\nChoose a cell from 1 to 9\n\n", currentPlayer)
+
+		// add empty lines to separate logical parts and make code more readable
+
 		choice := 0
 		fmt.Scanf("%d", &choice)
+
 		if choice < 1 || choice > 9 {
 			fmt.Println("Invalid choice!")
 			continue
 		}
-		if board.SetCellValue(choice-1, currentPlayer) {
-			if board.HasWinner(choice-1, currentPlayer) {
-				fmt.Printf("Game won by %c\n", currentPlayer)
-				return
-			}
-			if !board.HasEmptyCells() {
-				fmt.Println("Game is draw")
-				return
-			}
-			tmp := currentPlayer
-			currentPlayer = nextPlayer
-			nextPlayer = tmp
-		} else {
+
+		// always consider flat code over nested
+
+		if !b.MakeMove(choice-1, currentPlayer) {
 			fmt.Println("Invalid move!")
 			continue
 		}
 
-	}
+		if b.HasWinner() {
+			// print result
+			b.Print()
+
+			fmt.Printf("Game won by %c\n", currentPlayer)
+			return
+		}
 
+		if !b.HasEmptyCells() {
+			fmt.Println("Game is draw")
+			return
+		}
+
+		currentPlayer, nextPlayer = nextPlayer, currentPlayer // there is multiassignments in go
```

Comment:
I knew about this, but didn't realize that operations are happening "at the same time".

I thought it would unwind as:

  1. `currentPlayer = nextPlayer`
  2. `nextPlayer = currentPlayer`

Reply:
In go most of details works how you want they worked. And that's one more reason to love it.